gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -210,7 +216,20 @@ public GrpcCallContext withTransportChannel(TransportChannel inputChannel) {
           "Expected GrpcTransportChannel, got " + inputChannel.getClass().getName());
     }
     GrpcTransportChannel transportChannel = (GrpcTransportChannel) inputChannel;
-    return withChannel(transportChannel.getChannel());
+    return new GrpcCallContext(
+        transportChannel.getChannel(),
+        credentials,
+        transportChannel.isDirectPath() ? callOptions.withCallCredentials(null) : callOptions,
```

Comment:
Sorry, I meant can we do this logic/ check inside the private constructor? This should only pass the calloptions.

Constructor contains the:
`this.callOptions =  isDirectPath() ? calloptions.withCallCredentials(null) : callOptions` logic

Reply:
Ah this makes so much sense! Sorry I don't know what I was thinking. Changed.